Sources/ArcGISToolkit/Components/Augmented Reality/Utilities/GeoTrackingSceneView.swift

```diff
@@ -0,0 +1,199 @@
+        let cameraController = TransformationMatrixCameraController()
+        cameraController.translationFactor = 1
+        cameraController.clippingDistance = clippingDistance
+        _cameraController = .init(initialValue: cameraController)
+    }
+    
+    public var body: some View {
+        SceneViewReader { sceneViewProxy in
+            ZStack {
+                ARSwiftUIView(proxy: arViewProxy)
+                    .onDidUpdateFrame { _, frame in
+                        guard let interfaceOrientation, initialCameraIsSet else { return }
+                        
+                        sceneViewProxy.updateCamera(
+                            frame: frame,
+                            cameraController: cameraController,
+                            orientation: interfaceOrientation,
+                            initialTransformation: .identity
+                        )
+                        sceneViewProxy.setFieldOfView(
+                            for: frame,
+                            orientation: interfaceOrientation
+                        )
+                    }
+                    .onDidChangeGeoTrackingStatus { session, status in
+                        switch status.state {
+                        case .notAvailable, .initializing, .localizing:
+                            Task.detached { @MainActor in
+                                initialCameraIsSet = false
+                            }
+                        case .localized:
+                            // Update the camera controller every time geo-tracking is localized,
+                            // to ensure the best experience.
+                            if !initialCameraIsSet, let currentLocation, let currentHeading {
+                                // Set the initial heading of scene view camera based on location
+                                // and heading. Geo-tracking requires 90 degrees rotation.
+                                updateCameraController(
+                                    location: currentLocation,
+                                    heading: currentHeading + 90,
+                                    altitude: currentLocation.position.z ?? 0
+                                )
+                                Task.detached { @MainActor in
+                                    initialCameraIsSet = true
+                                }
+                            }
+                        @unknown default:
+                            fatalError("Unknown ARGeoTrackingStatus.State")
+                        }
+                    }
+                
+                if initialCameraIsSet {
+                    sceneViewBuilder(sceneViewProxy)
+                        .cameraController(cameraController)
+                        .attributionBarHidden(true)
+                        .spaceEffect(.transparent)
+                        .atmosphereEffect(.off)
+                        .interactiveNavigationDisabled(true)
```

Comment:
I assume this is duplicated down below in the world tracking version. We should try to take as much as we can and consolidate to avoid code duplication. For example, these 5 modifiers can be rolled into a single modifier.

Reply:
aa77704077e3ba7b69aa881a12357e3dbe1b984f , will need a better place to put it and a better name…